backend/plugin/mfa/otp/otp.go

```diff
@@ -0,0 +1,104 @@
+// Package otp is the plugin for OTP Multi-Factor Authentication.
+package otp
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pquerna/otp/totp"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	// issuerName is the name of the issuer of the OTP token.
+	issuerName = "Bytebase"
+	// secondsInMinute is the number of seconds in a minute.
+	secondsInMinute = 60
+	// maxPastSecretCount is the maximum number of past secret we will check.
+	maxPastSecretCount = 5
+)
+
+// removeSecondsFromTimestamp removes the seconds from the timestamp.
+func removeSecondsFromTimestamp(timestamp int64) int64 {
+	t := time.Unix(timestamp, 0)
+	truncated := time.Date(t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), 0, 0, t.Location())
+	return truncated.Unix()
+}
+
+// TimeBasedReader is a reader that returns the same value for the same timestamp.
+type TimeBasedReader struct {
+	reader *strings.Reader
+}
+
+// NewTimeBasedReader creates a new TimeBasedReader.
+func NewTimeBasedReader(timestamp int64) *TimeBasedReader {
```

Comment:
Pass in time.Time, or rename the argument to indicate the unit.

Reply:
How about rename `NewTimeBasedReader` to `NewTimeBasedReaderWithTimestamp`?

Reply:
Update, PTAL.